app/services/fare_charger_service.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class FareChargerService
+  def initialize(card)
+    @card = card
+  end
+
+  def charge_max
+    max_fare = Fare.maximum(:amount)
+    @card.balance -= max_fare
+    @card.transactions.create!(amount: -max_fare, description: 'Charge for max fare')
+    @card.save!
```

Comment:
Again here you can take advantage of the instance method defined on `Card` model to charge or add an amount to  given card. Also would be great to add the possibility to pass a description